Restructure parallelization and Caching documentation (#609)
aai-institute/pyDVL: docs/how-to-guides/how-to-scale-up-if-with-parallel.md

---
title: How-to scale up influence function with parallelization
alias:
    name: scale-up-if-with-parallel
---

Calculating influence functions on large datasets can be challenging due to
memory constraints (e.g. input data does not into memory or large models).
In these cases, we want to map our influence function model over
collections of batches (or chunks) of data.

This guide shows you how to parallelize influence function
computations using pyDVL to handle large batches of data efficiently.

!!! tip "Other guides"

    For speeding-up data valuation algorithms with parallelization
    see [[speed-up-value-with-parallel]].

> ### Sequential Computation

Reviewer: I think the sequential computation should be explained in a separate document that is marked as prerequisite for this one.

---
title: How-to scale up influence function with parallelization
alias:
    name: scale-up-if-with-parallel
---

Calculating influence functions on large datasets can be challenging due to
memory constraints (e.g. input data does not into memory or large models).
In these cases, we want to map our influence function model over
collections of batches (or chunks) of data.

This guide shows you how to parallelize influence function
computations using pyDVL to handle large batches of data efficiently.

!!! tip "Other guides"

    For speeding-up data valuation algorithms with parallelization
    see [[speed-up-value-with-parallel]].

!!! warning "Speeding up Influence Function"

>     Scaling up Influence Function computations will not speed up

Reviewer 2: Well, this is only true, if all data fits into memory and the additional overhead would not be needed. But this holds true in great generality, so I would not specifically mention it here.

Author: I removed it.

    the computations and may even slow them down.

### Sequential Computation

> The simplest way to compute influence functions is sequentially using the

Reviewer 2: ```suggestion
The simplest way to compute influence functions is using the convenience class 
```

[SequentialInfluenceCalculator][pydvl.influence.influence_calculator].

This uses a double for-loop to iterate over the batches sequentially
and collects them.

We start by instantiating the dataloaders and the model and then fitting
the latter:

```python
from pydvl.influence.torch import CgInfluence
from torch.utils.data import DataLoader

batch_size = 10
train_dataloader = DataLoader(..., batch_size=batch_size)
test_dataloader = DataLoader(..., batch_size=batch_size)

> infl_model = CgInfluence(model, loss, hessian_regularization=0.01)

Reviewer 2: ```suggestion
infl_model = CgInfluence(model, loss, regularization=0.01)
```

train_dataloader = DataLoader(train_data_set)

> infl_model = CgInfluence(model, loss, hessian_regularization=0.01)

Reviewer 2: ```suggestion
infl_model = CgInfluence(model, loss, regularization=0.01)
```
